src/mapmanager.rs

```diff
@@ -378,13 +378,20 @@ pub async fn switch_map_to(
 
     let _ = bf4.set_preset(Preset::Custom).await;
     let _ = bf4.set_vehicles_spawn_allowed(vehicles).await;
+
+    // Force the vehicle spawn delay to the default value, 100.
+    // We do this as a safeguard against previously seen quirks,
+    // where the value would be automatically set to 400.
+    let _ = bf4.set_vehicle_spawn_delay(100).await;
+
     let _  = dbg!(bf4.set_tickets(tickets).await);
     sleep(Duration::from_secs(1)).await;
 
     bf4.maplist_run_next_round().await?;
 
     sleep(Duration::from_secs(10)).await;
     let _  = dbg!(bf4.set_tickets(std::cmp::max(100, tickets)).await);
+    let _ = bf4.set_vehicle_spawn_delay(100).await;
```

Comment:
I don't think this needs to be set here again, since the value being 100 doesn't violate hardcore mode. Remove this line and see if it still works then?